### PR TITLE
Provide a clear EventSource for events of schedulers

### LINKED ISSUE #21969: Provide a clear EventSource for events of schedulers
```api.EventSource{Component: s.SchedulerName}``` in the event of schedulers only has SchedulerName, it's not very clear. Maybe we need to change it to ```api.EventSource{Component: "scheduler", SchedulerName: s.SchedulerName}```. 
ref #19724 
@mikedanese What's your point?

### PR DESCRIPTION
**Pull request by feihujiang**
Fix #21969

Change `api.EventSource{Component: s.SchedulerName}` to `api.EventSource{Component: "scheduler", SchedulerName: s.SchedulerName}`

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -2503,6 +2503,8 @@ type EventSource struct {
 	Component string `json:"component,omitempty"`
 	// Host name on which the event is generated.
 	Host string `json:"host,omitempty"`
+	// Scheduler name on which the event is generated.
+	SchedulerName string `json:"schedulerName,omitempty"`
```
Comment (lavalamp) [inline pkg/api/v1/types.go]: I don't think it makes sense to have a schedulerName field here. I think the scheduler name is actually part of the component field (e.g., "scheduler-batch"). If we really want to separate it, I think we should define something like shardName, instanceName, role, etc. Events are more general than scheduler and can't reference schedulers specifically.

Alternatively, we could add a "componentClass" field, set *that* to scheduler, and set the current component field to the name of the scheduler. "shard" may actually be a different thing entirely, like I can imagine wanting to say scheduler, batch, shard 3.

@davidopp may have opinions too. The only thing I feel strongly about is that we can't reference specific components like schedulers in this object.

Comment (davidopp) [inline pkg/api/v1/types.go]: Yeah, I definitely agree with @lavalamp on all points.

Concretely, we should just make Component include the scheduler name. In principle schedulers can be very different (written in different languages, using totally different scheduling rules as long as they obey the Kubelet admission rules, etc.), so they should just be considered totally independent components. If you run multiple replicas of the same scheduler you can just number them arbitrarily, like scheduler-01, scheduler-02, etc.

Comment (lavalamp) [inline pkg/api/v1/types.go]: Yeah, that's my first choice, too.

On Fri, Feb 26, 2016 at 3:57 PM, David Oppenheimer <notifications@github.com

Comment (feihujiang) [inline pkg/api/v1/types.go]: Thanks for detailed reasons. But I have a question, If component only include the scheduler name, and  the scheduler name is like aaa, or bbb, etc, we couldn't know events is from a scheduler. If I want to know whether events is from a scheduler, how could I do? Make a strict rule that the scheduler name should be prefixed with scheduler, like scheduler-aaa-01, scheduler-aaa-02?

Comment (lavalamp) [inline pkg/api/v1/types.go]: Yes, I'd recommend all schedulers have their component name prefixed with
"scheduler-".

Comment (feihujiang) [inline pkg/api/v1/types.go]: OK, I got it.

**Pull request closed without merging**